models/executor.js

```diff
@@ -1,8 +1,14 @@
+var uuid = require('uuid');
+
 module.exports = function(Executor) {
-  Executor.beforeCreate = function(next) {
-    if (this.remainingCapacity === -1) {
+  Executor.definition.properties.id.default = function() {
+    return uuid.v4();
+  };
+
+  Executor.beforeSave = function(callback) {
+    if (this.remainingCapacity === null && this.totalCapacity !== null) {
```

Comment:
This looks like initialization code, but it is in `beforeSave` instead of `beforeCreate`. Is this in place of a implementing `remainingCapacity.default()`?